### DISCUSSION TITLE
Feature Request: XY Plot / Add a third Metric in Grafana Plot to Group Results and have seperate auto generated colours in the Graph

### DISCUSSION THREAD

**Discussion by Bubbu17** [grafana/grafana#38382] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
I would like to have a third Metric to Group Results in the Graph by different colors and different labels which are shown in the Legend as well.

a snippet of data data source which are used, all data are written in a column a could only be separated by its unique ID Column, this would be the third metric I would prefer to use to Group the data.
![image](https://user-images.githubusercontent.com/62136070/82823134-5326cc00-9ea7-11ea-9d25-f4b23c72685e.png)
or
![image](https://user-images.githubusercontent.com/62136070/82823162-5e79f780-9ea7-11ea-8c95-6bb4b13c5a7d.png)

I was able to set up the graf and the proper legend, but i am unable to show the caracteristic curves of the devices inside the graf becouse ther is no datetime in x direction.
![image](https://user-images.githubusercontent.com/62136070/82823267-8b2e0f00-9ea7-11ea-8c20-88f1507e73e8.png)

**Why is this needed**:
To show data from an SQL server which is not a time series and would usualy represented in an XY Plot. I am missing this function in Grafana without installing any third party Plugin..

Sample Data with 3 colums (ID: DAte, Time, Serial#) (DATA: Value in Y) (time: integer from 2 to130 representing teh measuring value in X):
ID data time
May 13 2020 7:25AM 700005575319 -0.01922607 2
May 13 2020 7:28AM 700005575319 0.04119873 2
May 11 2020 7:58AM 700005575320 0.04943848 2
May 13 2020 7:31AM 700005575319 -0.0604248 2
May 13 2020 8:10AM 700005575319 -0.07141113 2
May 13 2020 7:34AM 700005575319 -0.09338379 2
May 15 2020 5:54AM 700005575319 0.1263428 2
May 11 2020 7:49AM 700005575318 0.1428223 2
May 13 2020 7:25AM 700005575319 0.002746582 3
May 13 2020 7:31AM 700005575319 -0.05493164 3
May 13 2020 7:28AM 700005575319 0.05767822 3
May 13 2020 8:10AM 700005575319 -0.0604248 3
May 11 2020 7:58AM 700005575320 0.07141113 3
May 13 2020 7:34AM 700005575319 -0.08239746 3
May 15 2020 5:54AM 700005575319 0.1318359 3
May 11 2020 7:49AM 700005575318 0.1538086 3
May 13 2020 7:25AM 700005575319 0.01373291 4
May 13 2020 7:31AM 700005575319 -0.04394531 4
May 13 2020 8:10AM 700005575319 -0.04943848 4
May 13 2020 7:34AM 700005575319 -0.06591797 4
May 13 2020 7:28AM 700005575319 0.06866455 4
May 11 2020 7:58AM 700005575320 0.09887695 4
May 15 2020 5:54AM 700005575319 0.1318359 4
May 11 2020 7:49AM 700005575318 0.1538086 4
May 13 2020 7:25AM 700005575319 0.03570557 5
May 13 2020 8:10AM 700005575319 -0.03845215 5
May 13 2020 7:31AM 700005575319 -0.04394531 5
May 13 2020 7:34AM 700005575319 -0.05493164 5
May 13 2020 7:28AM 700005575319 0.08514404 5
May 11 2020 7:58AM 700005575320 0.1098633 5
May 15 2020 5:54AM 700005575319 0.1428223 5
May 11 2020 7:49AM 700005575318 0.1593018 5
May 13 2020 8:10AM 700005575319 -0.02197266 6
May 13 2020 7:34AM 700005575319 -0.03295898 6
May 13 2020 7:31AM 700005575319 -0.04394531 6
May 13 2020 7:25AM 700005575319 0.05218506 6
May 13 2020 7:28AM 700005575319 0.09613037 6
May 11 2020 7:58AM 700005575320 0.1318359 6
May 15 2020 5:54AM 700005575319 0.1428223 6
May 11 2020 7:49AM 700005575318 0.1647949 6
May 13 2020 8:10AM 700005575319 -0.02197266 7
May 13 2020 7:34AM 700005575319 -0.03845215 7
May 13 2020 7:31AM 700005575319 -0.04943848 7
May 13 2020 7:25AM 700005575319 0.05767822 7
May 13 2020 7:28AM 700005575319 0.09063721 7
May 11 2020 7:58AM 700005575320 0.1318359 7
May 15 2020 5:54AM 700005575319 0.1318359 7
May 11 2020 7:49AM 700005575318 0.1538086 7
May 13 2020 8:10AM 700005575319 -0.01647949 8
May 13 2020 7:34AM 700005575319 -0.02746582 8
May 13 2020 7:31AM 700005575319 -0.03845215 8
May 13 2020 7:25AM 700005575319 0.06866455 8
May 13 2020 7:28AM 700005575319 0.09063721 8
May 15 2020 5:54AM 700005575319 0.1263428 8
May 11 2020 7:58AM 700005575320 0.1373291 8
May 11 2020 7:49AM 700005575318 0.1483154 8
May 13 2020 8:10AM 700005575319 -0.01647949 9
May 13 2020 7:34AM 700005575319 -0.02746582 9
May 13 2020 7:31AM 700005575319 -0.04394531 9
May 13 2020 7:25AM 700005575319 0.06866455 9
May 13 2020 7:28AM 700005575319 0.09063721 9
May 15 2020 5:54AM 700005575319 0.1098633 9
May 11 2020 7:49AM 700005575318 0.1318359 9
May 11 2020 7:58AM 700005575320 0.1428223 9
May 13 2020 8:10AM 700005575319 -0.01098633 10
May 13 2020 7:34AM 700005575319 -0.02746582 10
May 13 2020 7:31AM 700005575319 -0.03845215 10
May 13 2020 7:25AM 700005575319 0.06866455 10
May 13 2020 7:28AM 700005575319 0.09063721 10
May 15 2020 5:54AM 700005575319 0.09887695 10
May 11 2020 7:49AM 700005575318 0.1153564 10
May 11 2020 7:58AM 700005575320 0.1318359 10
May 13 2020 8:10AM 700005575319 -0.005493164 11
May 13 2020 7:34AM 700005575319 -0.02197266 11
May 13 2020 7:31AM 700005575319 -0.02746582 11
May 13 2020 7:25AM 700005575319 0.07965088 11
May 15 2020 5:54AM 700005575319 0.09338379 11
May 13 2020 7:28AM 700005575319 0.1016235 11
May 11 2020 7:49AM 700005575318 0.1153564 11
May 11 2020 7:58AM 700005575320 0.1318359 11
May 13 2020 8:10AM 700005575319 0.005493164 12
May 13 2020 7:34AM 700005575319 -0.01647949 12
May 13 2020 7:31AM 700005575319 -0.02197266 12
May 13 2020 7:25AM 700005575319 0.08514404 12
May 15 2020 5:54AM 700005575319 0.08789063 12
May 13 2020 7:28AM 700005575319 0.1071167 12
May 11 2020 7:49AM 700005575318 0.1098633 12
May 11 2020 7:58AM 700005575320 0.1318359 12
May 13 2020 8:10AM 700005575319 0.005493164 13
May 13 2020 7:31AM 700005575319 -0.01098633 13
May 13 2020 7:34AM 700005575319 -0.01647949 13
May 15 2020 5:54AM 700005575319 0.08239746 13
May 13 2020 7:25AM 700005575319 0.08514404 13
May 13 2020 7:28AM 700005575319 0.09613037 13
May 11 2020 7:49AM 700005575318 0.09887695 13
May 11 2020 7:58AM 700005575320 0.1318359 13
May 13 2020 7:31AM 700005575319 0 14
May 13 2020 7:34AM 700005575319 0 14
May 13 2020 8:10AM 700005575319 0.01647949 14
May 15 2020 5:54AM 700005575319 0.0769043 14
May 13 2020 7:25AM 700005575319 0.08514404 14
May 11 2020 7:49AM 700005575318 0.09887695 14
May 13 2020 7:28AM 700005575319 0.1071167 14
May 11 2020 7:58AM 700005575320 0.1318359 14
May 13 2020 7:34AM 700005575319 0 15
May 13 2020 7:31AM 700005575319 0.005493164 15
May 13 2020 8:10AM 700005575319 0.02197266 15
May 15 2020 5:54AM 700005575319 0.07141113 15
May 13 2020 7:25AM 700005575319 0.08514404 15
May 11 2020 7:49AM 700005575318 0.08789063 15
May 13 2020 7:28AM 700005575319 0.1126099 15
May 11 2020 7:58AM 700005575320 0.1373291 15
May 13 2020 7:34AM 700005575319 0.01098633 16
May 13 2020 7:31AM 700005575319 0.02197266 16
May 13 2020 8:10AM 700005575319 0.03295898 16
May 15 2020 5:54AM 700005575319 0.06591797 16
May 11 2020 7:49AM 700005575318 0.08239746 16
May 13 2020 7:25AM 700005575319 0.09063721 16
May 13 2020 7:28AM 700005575319 0.1290894 16
May 11 2020 7:58AM 700005575320 0.1538086 16
May 13 2020 7:31AM 700005575319 0.02746582 17
May 13 2020 7:34AM 700005575319 0.02746582 17
May 13 2020 8:10AM 700005575319 0.04943848 17
May 15 2020 5:54AM 700005575319 0.07141113 17
May 11 2020 7:49AM 700005575318 0.0769043 17
May 13 2020 7:25AM 700005575319 0.1071167 17
May 13 2020 7:28AM 700005575319 0.1345825 17
May 11 2020 7:58AM 700005575320 0.1757813 17
May 13 2020 7:31AM 700005575319 0.03295898 18
May 13 2020 7:34AM 700005575319 0.04394531 18
May 13 2020 8:10AM 700005575319 0.06591797 18
May 11 2020 7:49AM 700005575318 0.06591797 18
May 15 2020 5:54AM 700005575319 0.08239746 18
May 13 2020 7:25AM 700005575319 0.1290894 18
May 13 2020 7:28AM 700005575319 0.1455688 18
May 11 2020 7:58AM 700005575320 0.1922607 18

**Comment by Bubbu17** [comment]
Below the section where the third Metric, in my Case ID, should also be defined as a grouping operator for the curves. (this is a screenshot from Plottly which dosen't do the job neither)
![image](https://user-images.githubusercontent.com/62136070/82823684-4eaee300-9ea8-11ea-9842-f6698330fbcf.png)
![image](https://user-images.githubusercontent.com/62136070/82823811-83229f00-9ea8-11ea-87df-2127faace050.png)

**Comment by Bubbu17** [comment]
the result should lok like this:
![image](https://user-images.githubusercontent.com/62136070/82824866-69825700-9eaa-11ea-8c3c-b74d6772cfd7.png)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
